{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "de16b538",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(sumFREGAT)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d6a12f0b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene A2M, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene A2ML1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ABCA2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ACSM4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene AGAP4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene AICDA, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ALYREF, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene AMY1B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene AMY1C, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene ANKRD20A1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ANXA2R, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene ANXA8, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene ANXA8L1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene APOBEC1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene ARL17A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene ARL17B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ARL6IP4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ARRDC1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ATN1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ATOH1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene BAGE2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene BHLHA9, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene C12orf57, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene C1orf122, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene C1RL, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene C1S, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene C3AR1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene C4B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene C8G, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene C9orf139, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CAMP, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene CBWD3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene CBWD5, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene CBWD6, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene CCL3L3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene CCL4L1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene CCL4L2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CD163, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CD163L1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CD69, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CDCA3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CDRT15L2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CHRFAM7A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CHST14, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CLDN17, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CLEC2D, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CLEC4A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CLEC4C, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CLEC4D, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CLEC4E, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CLEC6A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CLECL1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CLIC3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CLSTN3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene CSH1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene CTAGE15, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene CTAGE4, skipped”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene CYSRT1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene DEFA1B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene DEFB104A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene DEFB104B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene DEFB106A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene DEFB106B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene DEFB107A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene DEFB107B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene DEFB121, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene DEFB4A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene DEFB4B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene DPH7, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene DPP7, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene DPPA3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene DRAP1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene EHMT1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene EMG1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ENO2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ENTPD2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ENTPD8, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene FAM157B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene FAM166A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene FAM229A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene FAM25C, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene FAM72B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene FAM72D, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene FAM86B1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene FAM86B2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene FAM90A1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene FBXW5, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene FOXD4L3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene FOXD4L5, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene FOXD4L6, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene FOXJ2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene FRG2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene FRG2B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene FRG2C, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene FUT7, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene GDF3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene GGT2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene GK2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene GOLGA6L1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene GOLGA6L10, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene GOLGA6L22, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene GOLGA6L6, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene GOLGA6L9, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene GOLGA8K, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene GOLGA8O, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene GOLGA8R, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene GTF2H2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene GTSF1L, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene HBA2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene HCRT, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene HGC6.3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene HMX1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene IFNA2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene IFNB1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene IL23A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene INO80B, skipped”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene KIR2DL1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene KLRB1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene KLRF1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene KLRG1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene KRTAP13-1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene KRTAP19-3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene KRTAP19-4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene KRTAP21-2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene KRTAP2-4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene KRTAP5-11, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene KRTAP6-1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene KRTAP9-6, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene KRTAP9-7, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene LCE2C, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene LCN12, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene LCNL1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene LOC100505502, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene LOC100996413, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene LOC100996643, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene LOC101928195, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene LOC101929583, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene LOC391322, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene LOC645202, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene LOC728715, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene LPCAT3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene LRRC23, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene LRRC26, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene LTB, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene M6PR, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene MBD3L4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene MBD3L5, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene MFAP5, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene MIA2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene MIEN1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene MMP23B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene MRPL41, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene MSMP, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene MT1HL1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NANOG, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NANOGNB, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NBPF1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene NBPF11, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NBPF20, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NBPF8, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NDOR1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene NDUFAF3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NECAP1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NELFB, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NKX2-4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NOG, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NOXA1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NPDC1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene NPIPA2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene NPIPB3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene NPIPB5, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NR2C2AP, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NRARP, skipped”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene NSMF, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene NUTM2B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene OR11H1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene OR11H12, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene OR11H2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR2A12, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene OR2A42, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR2AP1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR2AT4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR2T2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene OR2T29, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene OR2T3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene OR2T34, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene OR2T35, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene OR2T5, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR3A1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene OR4F17, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene OR4F4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene OR4F5, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene OR4M2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR4N2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR4X2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR5AN1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR5M8, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR5T3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR6C3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR6C65, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR6K2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR8B8, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR8K5, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR9A2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene OR9I1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PDE4DIP, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene PEX5, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene PHB2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene PHC1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene PLEKHG7, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene PMCH, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene POTEB, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene POTEB2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene POTED, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene POTEG, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene POTEH, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene POTEM, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene POU4F3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PPIAL4A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PPIAL4E, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PPIAL4G, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene PPP1R35, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRAMEF10, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRAMEF11, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRAMEF14, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRAMEF15, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRAMEF19, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRAMEF20, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRAMEF22, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRAMEF25, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRAMEF26, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRAMEF5, skipped”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRAMEF6, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRAMEF7, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRAMEF9, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRR20A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PRR20B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene PTGDS, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene PTPN20, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene PTPN6, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene PZP, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene RASA4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene RBP5, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene RGPD2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene RGPD6, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene RGPD8, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene RIMBP3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene RIMBP3B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene RIMKLB, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene RNASE1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene RNASE8, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene RNF208, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene RPL10L, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene RPL21, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene RPP25, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene RPS15, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene RPS17, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene RSPH10B2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene RXFP4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene SF3B5, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene SLC2A14, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene SLC2A3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene SLC34A3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene SMN1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene SMN2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene SPAG11B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene SPATA31A3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene SPSB2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene SRGAP2B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene SSNA1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene SULT1A3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene SULT1A4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TAAR8, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TAS2R10, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TAS2R40, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene TBC1D3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene TBC1D3B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene TBC1D3C, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene TBC1D3F, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene TBC1D3H, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene THOC3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TLX2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TMEM139, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene TMEM191B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TMEM191C, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TMEM203, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TMEM208, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TMEM210, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TOR4A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene TP53TG3, skipped”\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TPI1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TPRN, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TPTE, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TRAM1L1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TRIM49D2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TRIM73, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TRMT112, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene TUBB4B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene UAP1L1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene UBL5, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene UCN, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene USP17L1, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene USP17L10, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene USP17L15, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene USP17L18, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene USP17L2, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene USP17L29, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene USP17L3, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene USP17L4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene USP17L7, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene USP17L8, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene VN1R4, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ZMYND19, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene ZNF26, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ZNF705A, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene ZNF705B, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No variants to analyze in gene ZNF705D, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ZNF705G, skipped”\n",
      "Warning message in get.sumstat(score.file, gene.file, gene, anno.type, cor.path, :\n",
      "“No matching variants in correlation file for gene ZNHIT2, skipped”\n"
     ]
    }
   ],
   "source": [
    "SKATO(score.file = 'WLS.vcf.gz', gene.file = \"refFlat_uniq_filtered.txt\", cor.path = './SNP_cor_matrix/all',\n",
    "     write.file = 'results/SKATO.txt', quiet = FALSE, beta.par=c(1,1))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:R]",
   "language": "R",
   "name": "conda-env-R-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
